packages/rulesets/generated/spectral/az-arm.js

```diff
@@ -2418,6 +2418,21 @@ const withXmsResource = (putOperation, _opts, ctx) => {
     return errors;
 };
 
+const errorMessage = "If Properties with type:object dont have a reference model defined, then the allowed types can only be primitive data types.";
+const propertiesTypeObjectNoDefinition = (definitionObject, _opts, ctx) => {
+    if ((definitionObject === null || definitionObject === void 0 ? void 0 : definitionObject.type) === "") {
+        return [{ message: errorMessage, path: ctx.path }];
```

Comment:
>    return [{ message: errorMessage, path: ctx.path }]; [](http://example.com/codeflow?start=4&length=55)

Inconsistent tabs. Installing the "Prettier - Code Formatter" VS code extension helped me fix styling issues.